src/string_vector_serializer.cpp

```diff
@@ -0,0 +1,52 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "string_vector_serializer.h"
+
+namespace paraglob {
+  // Serializes a vector of strings into a string.
+  // Prepends the length of each string followed by a seperator to the output
+  // string and then appends all the strings in the vector
+  std::string serialize_string_vec(const std::vector<std::string>& in) {
+    std::string delimiter ("~~~~~/////~~~~~");
+    std::string nums ("");
+    std::string out ("");
+
+    for (std::string s : in) {
+      nums = nums + std::to_string(s.length()) + ",";
+      out = out + s;
+    }
+
+    return (nums + delimiter + out);
+  }
+
+  std::vector<std::string> unserialize_string_vec(std::string serialized) {
```

Comment:
Pass by const reference

Reply:
The method is currently mutating the string as it moves through it, but given the above comment I agree that this should just pass by const reference.